### ISSUE TITLE
Port to GNOME 47

### ISSUE THREAD

**Issue by idoric** [neuromorph/openbar#74]
GNOME 47 has been released yesterday. Please update for GNOME 47 compatibility. The upgrade guide can be found here: [Port Extensions to GNOME Shell 47](https://gjs.guide/extensions/upgrading/gnome-shell-47.html)

Thanks!

**Comment by neuromorph** [comment]
Gnome 47 support will be included in the next release coming soon.
It is already available in GitHub `main` branch if you want to do a manual install. 
I will try it out for a bit to iron out any wrinkles before official release. If you see anything that needs tweaking, let me know.

An e.g. -
Gnome has changed the style for dialog buttons. It used to stick to bottom edge of dialog and so OpenBar styles had to follow dialog corner radius. In 47, they have moved the buttons inside, not sure if that's better. Now, OpenBar still does the borders as before and looks like this. It actually looks alright but need to decide whether to keep it or modify conditionally for 47.

![image](https://github.com/user-attachments/assets/227bf96e-d6f0-433e-8278-d43ab2a042b4)

**Comment by neuromorph** [comment]
Also, since Gnome 47 has introduced multiple Accent colors to choose from, here's how it works:

OpenBar already allows to set any arbitrary color as Accent color so this does not really add anything to it. You need to set Accent color from OpenBar settings as usual. The Accent set in OpenBar will take precedence over anything selected in Gnome settings. OpenBar as usual will apply the accent color to the entire shell, if enabled in settings.

There are, however, some artifacts that are graphics/animations in the Gnome Settings app e.g. Multitasking or Mouse/Touchpad. They also use accent color but not dynamically, the artifacts need to be created for supporting colors. Additionally, some may not choose to enable theming entire shell with OpenBar. For such reasons, whenever you set an accent color in OpenBar, it will automatically also select the closest accent color from Gnome and set that as Gnome Accent to make those artifacts and shell theme match as close to OpenBar as possible. 

TL;DR
Just set the Accent in OpenBar, corresponding Gnome Accent color will be auto-set in Gnome settings.

**Comment by neuromorph** [comment]
New version v41 with support for Gnome 47 is now live!

If you find any issue with the new version, feel free to log a bug report.
